java/src/com/ibm/streamsx/rest/internal/StandaloneAuthenticator.java

```diff
@@ -0,0 +1,193 @@
+/*
+# Licensed Materials - Property of IBM
+# Copyright IBM Corp. 2019 
+ */
+package com.ibm.streamsx.rest.internal;
+
+import static com.ibm.streamsx.topology.internal.gson.GsonUtilities.jstring;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.function.Function;
+
+import org.apache.http.client.fluent.Executor;
+import org.apache.http.client.fluent.Request;
+import org.apache.http.entity.ContentType;
+
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.annotations.Expose;
+import com.ibm.streamsx.topology.internal.streams.Util;
+
+public class StandaloneAuthenticator implements Function<Executor,String> {
+    private String resourcesUrl;
+    private String userName;
+    private String password;
+    private String securityUrl;
+    private String auth;
+    private long expire;    // in ms since epoch
+
+    private JsonObject cfg;
+
+    private static final String STREAMS_REST_RESOURCES = "/streams/rest/resources";
+    private static final String RESOURCES = "resources";
+    private static final String ACCESS_TOKENS = "accessTokens";
+    private static final String AUDIENCE_STREAMS = "{\"audience\":[\"streams\"]}";
+    private static final String EXPIRE_TIME = "expireTime";
+    private static final String ACCESS_TOKEN = "accessToken";
+    private static long MS = 1000;
+
+    public static StandaloneAuthenticator of(String endpoint, String userName, String password) {
+        if (endpoint == null) {
```

Comment:
Seems strange that this internal class is checking for environment variables. Won't the value(s) be always set by the `ofEndpoint` method?

Reply:
In the current usage, yes. I noticed that `ICP4DAutenticator.of()` was checking env vars because in the build / submit path it sometimes gets called with `null` for various params, and I wasn't sure if standalone would end up doing something similar.

I'm fine with removing those checks though and worrying about the build / submit path later if you like.